sdk/python/kfp/compiler/_default_transformers.py

```diff
@@ -87,8 +87,9 @@ def _add_name_for_oob_components(task):
         if hasattr(task, '_component_ref'):
             component_ref = task._component_ref.to_dict()
             origin_uri = component_ref.get('url')
```

Comment:
JFYI: `component_ref` always has the `.url` attribute. It's part of the structure. (It can be None though)

If your concern is linting, we'll be able to fix that when dropping support for Python 3.5.

Reply:
Fixed.